src/ledger-bridge-keyring.ts

```diff
@@ -0,0 +1,918 @@
+ *
+ * @param tx - Transaction to check, instance of either ethereumjs-tx or @ethereumjs/tx.
+ * @returns Returns `true` if tx is an old-style ethereumjs-tx transaction.
+ */
+function isOldStyleEthereumjsTx(
+  tx: TypedTransaction | OldEthJsTransaction,
+): tx is OldEthJsTransaction {
+  return 'getChainId' in tx && typeof tx.getChainId === 'function';
+}
+
+/**
+ * Check if the given payload is a SignTransactionPayload.
+ *
+ * @param payload - IFrame message response payload to check.
+ * @returns Returns `true` if payload is a SignTransactionPayload.
+ */
+function isSignTransactionResponse(
+  payload: IFrameMessageResponsePayload,
+): payload is SignTransactionPayload {
+  return hasProperty(payload, 'v') && typeof payload.v === 'string';
+}
+
+/**
+ * Check if the given payload is a SignMessagePayload.
+ *
+ * @param payload - IFrame message response payload to check.
+ * @returns Returns `true` if payload is a SignMessagePayload.
+ */
+function isSignMessageResponse(
+  payload: IFrameMessageResponsePayload,
+): payload is SignMessagePayload {
+  return hasProperty(payload, 'v') && typeof payload.v === 'number';
+}
+
+/**
+ * Check if the given payload is a GetAddressPayload.
+ *
+ * @param payload - IFrame message response payload to check.
+ * @returns Returns `true` if payload is a GetAddressPayload.
+ */
+function isGetAddressMessageResponse(
+  payload: IFrameMessageResponsePayload,
+): payload is GetAddressPayload {
+  return (
+    hasProperty(payload, 'publicKey') && typeof payload.publicKey === 'string'
+  );
+}
+
+/**
+ * Check if the given payload is a ConnectionChangedPayload.
+ *
+ * @param payload - IFrame message response payload to check.
+ * @returns Returns `true` if payload is a ConnectionChangedPayload.
+ */
+function isConnectionChangedResponse(
+  payload: IFrameMessageResponsePayload,
+): payload is ConnectionChangedPayload {
+  return (
+    hasProperty(payload, 'connected') && typeof payload.connected === 'boolean'
+  );
+}
+
+export class LedgerBridgeKeyring extends EventEmitter {
+  static type: string = keyringType;
+
+  readonly type: string = keyringType;
+
+  page = 0;
+
+  perPage = 5;
+
+  unlockedAccount = 0;
+
+  accounts: readonly string[] = [];
+
+  accountDetails: Record<string, AccountDetails> = {};
+
+  hdk = new HDKey();
+
+  hdPath = hdPathString;
+
+  paths: Record<string, number> = {};
+
+  network: NetworkApiUrls = NetworkApiUrls.Mainnet;
+
+  implementFullBIP44 = false;
+
+  iframeLoaded = false;
+
+  isDeviceConnected = false;
+
+  currentMessageId = 0;
+
+  messageCallbacks: Record<number, (response: IFrameMessageResponse) => void> =
+    {};
+
+  bridgeUrl: string = BRIDGE_URL;
+
+  iframe?: HTMLIFrameElement;
+
+  delayedPromise?: {
+    resolve: (value: unknown) => void;
+    reject: (error: unknown) => void;
+    transportType: string;
+  };
+
+  #eventListener?: (params: {
+    origin: string;
+    data: IFrameMessageResponse;
+  }) => void;
+
+  constructor(opts = {}) {
```

Comment:
Should this use `Partial<LedgerBridgeKeyringOptions>` as well? At the moment it seems to be lacking any specific type